rfcs/experimental/flow_graph_wait_try_put_and_wait/README.md

```diff
@@ -0,0 +1,510 @@
+never be decreased.
+
+It is impossible to rely on the number of successors while making a decision to store the metainformation in the buffer since if the node is used as part of the push-pull protocol,
+the number of successors is also equal to `0` since the edge is reversed.
+
+Current proposal considers this scenario is a `try_put_and_wait` feature limitation and does not add any support for that.
+
+### Multi-output nodes support
+
+Multi-output nodes (`multifunction_node` and `async_node`) creates an extra issue for the wait-for-one feature. Consider the following example (all of the examples shown
+in this section are for `multifunction_node` but also affects `async_node` in the same manner):
+
+```cpp
+
+using mf_node_type = multifunction_node<int, std::tuple<int, int>>;
+using output_ports = typename mf_node_type::output_ports_type;
+
+mf_node_type mfn(g, unlimited,
+    [](int input, output_ports& ports) {
+        std::get<0>(ports).try_put(output);
+        std::get<1>(ports).try_put(output);
+    })
+
+```
+
+Unlike the `function_node`, the computed output is not propagated implicitly to the successors. It is done explicitly by the user by calling
+`try_put(output)` on the corresponding output_port of the node. Since the metainformation associated with the particular input is hidden from the user
+inside of the Flow Graph implementation, it cannot be propagated to the successors as part of the explicit user try_put.
+
+An other interesting use-case is when the `multifunction_node` is used as a reduction for multiple input values with only one output. In that case the metainformation
+should not be automatically propagated at all and should be accumulated simultaneously with the outputs.
+
+Current implementation does not support ``multifunction_node`` and ``async_node``. 
+
+Possible approaches for implementing such support:
+
+* Hiding metainfo inside of ``output_ports`` to preserve automatic metainfo propagation.
+* Merging the input and the ``message_metainfo`` together into some publicly available type ``tagged_input<T>`` and require the user to explicitly specify this type
+for multi-output nodes.
+* Introducing an extra unspecified type ``node_type::tag_type`` and require the user to accept it as a third argument of the body. 
+
+## Open Questions
+
+The following questions should be addressed before moving this feature to ``supported``:
+
+* Multi-output nodes support should be described and implemented
+* Since the buffered item holds an additional reference counter on the associated metainfo, all elements should be retrieved from buffers to allow ``try_put_and_wait`` to exit.
+    * Should ``clear()`` member function be added to all of the buffering nodes in the graph (currently supported only in ``write_once_node`` and ``overwrite_node``).
+    * Should ``clear()`` member function be added to the non-rejecting ``join_node`` to handle the case when when we don't have the input present on each input port.
```

Comment:
I would instead consider something like `try_get_partial` which would return e.g. a tuple of optionals. It would allow draining the node without losing stored messages.

Reply:
Added